app/src/main/java/io/gnosis/safe/ui/StartActivity.kt

```diff
@@ -303,19 +307,21 @@ class StartActivity : BaseActivity(), SafeOverviewNavigationHandler, AppStateLis
     }
 
     override fun appInForeground() {
+        if (settingsHandler.requirePasscodeToOpen && settingsHandler.usePasscode && comingFromBackground) {
+            askForPasscode()
+        }
+        comingFromBackground = false
     }
 
     override fun appInBackground() {
         comingFromBackground = true
-        if (settingsHandler.requirePasscodeToOpen && settingsHandler.usePasscode && comingFromBackground) {
-            askForPasscode()
-            comingFromBackground = false
-        }
     }
 
     private fun askForPasscode() {
-        Navigation.findNavController(this@StartActivity, R.id.nav_host).navigate(R.id.enterPasscodeFragment, Bundle().apply {
-            putBoolean("requirePasscodeToOpen", true)
-        })
+        handler.postDelayed({
+            Navigation.findNavController(this@StartActivity, R.id.nav_host).navigate(R.id.enterPasscodeFragment, Bundle().apply {
+                putBoolean("requirePasscodeToOpen", true)
+            })
+        }, 600)
```

Comment:
this delay here is only needed if askForPasscode is called in onCreate before handleIntent. So 
```kotlin
if (settingsHandler.requirePasscodeToOpen && settingsHandler.usePasscode) {
            askForPasscode()
}
could be moved to handleIntent after all other navigation activities. I can imagine 600ms is unreliable value and we can't garantee all transitions will happen withing this interval. When comming from background this delay is not needed at all.
```

Reply:
The delay in mainly needed when opening from a push notification. But also when coming back from background. When I put this into handleIntent.  Then askForPasscode() is called twice when opening from a push notification.